### PR TITLE
MSSPeak Enhancement

### PR DESCRIPTION
**Pull request by joshpatten**
Commit to add MSSpeak to Newfies-Dialer

### CODE REVIEW

File: lua/libs/texttospeech.lua [hunk 1]
```diff
@@ -195,5 +195,17 @@ function tts(text, tts_dir)
         output_file = tts_acapela:run()
     end
 
+    elseif TTS_ENGINE == 'msspeak' then
+        -- Microsoft Speak
+        MSSpeak = require "msspeak"
+		tts_msspeak = MSSpeak:new{
+            APPLICATION_LOGIN=APPLICATION_LOGIN,
+            APPLICATION_PASSWORD=APPLICATION_PASSWORD,
```
Comment (areski) [inline lua/libs/texttospeech.lua]: let use the orignal name client_id and client_secret

File: lua/libs/settings.lua [hunk 2]
```diff
@@ -9,19 +9,36 @@ DBPASS = 'password'
...
 --
 ACCOUNT_LOGIN = 'EVAL_VAAS'
 APPLICATION_LOGIN = 'EVAL_YYYYYYY'
 APPLICATION_PASSWORD = 'XXXXXXXX'
 
+--
+-- For MSSpeak comment out Acapela SERVICE_URL
+-- and uncomment other SERVICE_URL parameter
+--
+-- SERVICE_URL = 'http://api.microsofttranslator.com/V2/Http.svc/Speak'
```
Comment (areski) [inline lua/libs/settings.lua]: let's try to avoir service_url for MSTranslate, and use the default value, this is likely not to change for a long time

File: lua/libs/texttospeech.lua [hunk 3]
```diff
@@ -195,5 +195,17 @@ function tts(text, tts_dir)
...
+            SERVICE_URL=SERVICE_URL,
```
Comment (areski) [inline lua/libs/texttospeech.lua]: let s remove service_url and use the default

File: lua/libs/texttospeech.lua [hunk 4]
```diff
@@ -195,5 +195,17 @@ function tts(text, tts_dir)
         output_file = tts_acapela:run()
     end
 
+    elseif TTS_ENGINE == 'msspeak' then
```
Comment (areski) [inline lua/libs/texttospeech.lua]: TTS_ENGINE == 'mstranslator' as this is the correct name for their service.

File: lua/libs/texttospeech.lua [hunk 5]
```diff
@@ -195,5 +195,17 @@ function tts(text, tts_dir)
...
+        -- Microsoft Speak
+        MSSpeak = require "msspeak"
+		tts_msspeak = MSSpeak:new{
+            APPLICATION_LOGIN=APPLICATION_LOGIN,
```
Comment (areski) [inline lua/libs/texttospeech.lua]: same commennt as above for login and password use client_id and client_secret

File: newfies/newfies_dialer/settings.py [hunk 6]
```diff
@@ -534,13 +534,20 @@
 
 # TEXT-TO-SPEECH
 # ==============
-TTS_ENGINE = 'FLITE'  # FLITE, CEPSTRAL, ACAPELA
+TTS_ENGINE = 'FLITE'  # FLITE, CEPSTRAL, ACAPELA, MSSPEAK
```
Comment (areski) [inline newfies/newfies_dialer/settings.py]: MSTRANSLATOR

File: newfies/newfies_dialer/settings.py [hunk 7]
```diff
@@ -534,13 +534,20 @@
...
 
+# ACAPELA AND MSSPEAK SETTINGS
+# MSSPEAK only needs APPLICATION_LOGIN(client_id)
+# and APPLICATION_PASSWORD(client_secret).
+# ACCOUNT_LOGIN is ignored for MSSpeak
 ACCOUNT_LOGIN = 'EVAL_XXXX'
 APPLICATION_LOGIN = 'EVAL_XXXXXXX'
 APPLICATION_PASSWORD = 'XXXXXXXX'
```
Comment (areski) [inline newfies/newfies_dialer/settings.py]: let s make a section for MSTRANSLATOR and not mix ACAPELA with it, no need

File: newfies/survey/function_def.py [hunk 8]
```diff
@@ -13,6 +13,22 @@
...
+    Run Microsoft Speak Text2Speech and return audio url
+    """
+    import msspeak
+    DIRECTORY = settings.MEDIA_ROOT + '/tts/'
+    if not tts_language:
+        tts_language = 'en'
+    tts_msspeak = msspeak.MSSpeak(
+        settings.APPLICATION_LOGIN,
+        settings.APPLICATION_PASSWORD,
+        DIRECTORY)
```
Comment (areski) [inline newfies/survey/function_def.py]: comments same as above

File: newfies/survey/views.py [hunk 9]
```diff
@@ -516,6 +517,9 @@ def section_script_play(request, id):
         if settings.TTS_ENGINE == 'ACAPELA':
             # Acapela
             audio_file_path = settings.MEDIA_ROOT + '/' + getaudio_acapela(script_text, 'US')
+        elif settings.TTS_ENGINE == 'MSSPEAK':
```
Comment (areski) [inline newfies/survey/views.py]: MSTRANSLATOR

**Comment by joshpatten** [comment]
I'll try again.

**Comment by joshpatten** [comment]
Well, I almost had everything cleaned up and ready for a new pull :-)

Let me know if I made any major screwups, I'll fix them and create a new PR.

**Comment by joshpatten** [comment]
Attached

**Pull request merged**